### PR TITLE
test(e2e): fix proposer selection test

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

e2e tests fail due to new proposer selection algorithm

## What was done?

1. Fixed proposer selection logic in `validator_test.go` to support consensus version 0 and 1 algorithm
2. Implemented support for changing consensus version in e2e app
3. Run TestApp_TxTooBig on just one node, to avoid timeouts

In future, we might also introduce vote rejection logic to e2e tests, to emulate increasing rounds.

## How Has This Been Tested?

CI/CD passed.

## Breaking Changes

None

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced consensus version updates functionality in the configuration, enhancing management of consensus parameters.
	- Added new sections in configuration files for validator and consensus version updates.

- **Bug Fixes**
	- Improved logic for validator selection and block proposing in tests, enhancing robustness.

- **Tests**
	- Enhanced test logic for processing consensus version updates, ensuring efficiency and accuracy. 
	- Implemented limits on block fetching during tests to prevent excessive durations.

### CODE REVIEW

**Comment by lklimek** [comment]
@coderabbitai review

File: test/e2e/tests/app_test.go [hunk 1]
```diff
@@ -242,68 +242,70 @@ func TestApp_TxTooBig(t *testing.T) {
...
-		// FIXME: ConsensusParams is broken for last height, this is just workaround
-		status, err := client.Status(ctx)
-		assert.NoError(t, err)
-		cp, err := client.ConsensusParams(ctx, &status.SyncInfo.LatestBlockHeight)
-		assert.NoError(t, err)
+	// FIXME: ConsensusParams is broken for last height, this is just workaround
+	status, err := client.Status(ctx)
+	assert.NoError(t, err)
+	cp, err := client.ConsensusParams(ctx, &status.SyncInfo.LatestBlockHeight)
+	assert.NoError(t, err)
```
[the remark this answers is not in the document]
Comment (lklimek) [reply test/e2e/tests/app_test.go:274]: ticket already there

File: test/e2e/tests/e2e_test.go [hunk 2]
```diff
@@ -122,6 +126,10 @@ func fetchBlockChain(ctx context.Context, t *testing.T) []*types.Block {
 
 	from := status.SyncInfo.EarliestBlockHeight
 	to := status.SyncInfo.LatestBlockHeight
+	// limit blocks to fetch to avoid long test times
+	if to-from > maxBlocks {
```
Comment (shumkov) [inline test/e2e/tests/e2e_test.go:130]: Do you expect any issues if we sync less than `LatestBlockHeight`? Maybe print a warning here so you'll now that we didn't sync completely?

Comment (lklimek) [reply test/e2e/tests/e2e_test.go:130]: No issues expected in e2e tests, we rarely go to > 100 blocks.

It does make sense on local testing, using long-running e2e network (but this is really rarely used, obly when debugging)

**Pull request merged**